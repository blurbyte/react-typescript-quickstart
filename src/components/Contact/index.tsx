import * as React from 'react';

import SecondaryHeadline from '../SecondaryHeadline';
import TextList from '../TextList';
import ExternalLink from '../ExternalLink';

class Contact extends React.Component {
  shouldComponentUpdate() {
    return false;
  }

  render() {
    return (
      <section>
        <SecondaryHeadline>Check it out!</SecondaryHeadline>
        <TextList simple>
          <TextList.Item>
            Check out my other projects on <ExternalLink href="https://github.com/blurbyte">GitHub</ExternalLink>
          </TextList.Item>
          <TextList.Item>
            Contact me via <ExternalLink href="https://twitter.com/blurbyte">Twitter</ExternalLink>
          </TextList.Item>
          <TextList.Item>
            Visit <ExternalLink href="http://blurbyte.com">blurbyte.com</ExternalLink>
          </TextList.Item>
        </TextList>
      </section>
    );
  }
}

export default Contact;
