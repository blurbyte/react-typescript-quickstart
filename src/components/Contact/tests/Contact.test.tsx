import * as React from 'react';
import { shallow } from 'enzyme';

import Contact from '../index';

describe('Contact', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Contact />);
    expect(wrapper).toMatchSnapshot();
  });
});
