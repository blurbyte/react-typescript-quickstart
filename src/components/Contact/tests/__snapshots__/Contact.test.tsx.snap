// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders correctly 1`] = `
<section>
  <styled.h2>
    Check it out!
  </styled.h2>
  <TextList
    simple={true}
  >
    <Styled(styled.li)>
      Check out my other projects on 
      <styled.a
        href="https://github.com/blurbyte"
      >
        GitHub
      </styled.a>
    </Styled(styled.li)>
    <Styled(styled.li)>
      Contact me via 
      <styled.a
        href="https://twitter.com/blurbyte"
      >
        Twitter
      </styled.a>
    </Styled(styled.li)>
    <Styled(styled.li)>
      Visit 
      <styled.a
        href="http://blurbyte.com"
      >
        blurbyte.com
      </styled.a>
    </Styled(styled.li)>
  </TextList>
</section>
`;
